libs/ebpfnetsh/programs.cpp

```diff
@@ -180,9 +213,11 @@ handle_ebpf_delete_program(
         return status;
     }
 
-    // TODO(issue #190): If the program is pinned, unpin the specified program.
-    // The temporary API ebpf_api_unpin_object requires knowing the name a priori
-    // and we have no way to get it yet.
+    // If the program is pinned, unpin the specified program.
+    status = _unpin_program_by_id(id);
```

Comment:
can we save some processing by first checking if the object is indeed pinned?

Reply:
there's not a more efficient way to check... yet.   There should be in the PR that will address #188